contracts/permissionless/OrderbookReserve.sol

```diff
@@ -408,13 +425,10 @@ contract OrderbookReserve is OrderIdManager, Utils2, KyberReserveInterface, Orde
     function withdrawKncFee(uint amount) public {
 
         address maker = msg.sender;
-        uint makerFreeAmount = uint(makerKnc[maker].freeKnc);
 
-        require(makerFreeAmount >= amount);
+        require(makerUnlockedKnc(maker) >= amount);
```

Comment:
you used to have a require to check the casting: require(uint(uint128(amount)) == amount);
maybe keep it?